component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -887,9 +887,30 @@ private void redirectToErrorPage(HttpServletResponse response, AuthenticationCon
     protected void processEmailOTPFlow(HttpServletRequest request, HttpServletResponse response, String email,
                                      String username, String queryParams,
                                      AuthenticationContext context) throws AuthenticationFailedException {
+
         Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
         Map<String, String> emailOTPParameters = getAuthenticatorConfig().getParameterMap();
+        boolean showAuthFailureReason =
+                Boolean.parseBoolean(emailOTPParameters.get(EmailOTPAuthenticatorConstants.SHOW_AUTH_FAILURE_REASON));
         try {
+            if (isLocalUser(context) &&
+                    EmailOTPUtils.isAccountLocked(getAuthenticatedUser(context))) {
+                String retryParam;
+                if (showAuthFailureReason) {
+                    String unlockTime = getUnlockTime(getAuthenticatedUser(context));
```

Comment:
Can this return null? If so shall we add a log more context to identify that?

Reply:
fixed in be18eca